build.rs

```diff
@@ -1,34 +1,156 @@
     } else {
-        "2.0.0"
+        ("2.0.0", "v2.0")
     };
-    let lib = pkg_config::Config::new()
-        .atleast_version(required_version)
-        .statik(true)
-        .probe("hwloc")
-        .expect("Could not find a suitable version of hwloc");
-    if cfg!(target_family = "unix") {
-        for link_path in lib.link_paths {
-            println!(
-                "cargo:rustc-link-arg=-Wl,-rpath,{}",
-                link_path
-                    .to_str()
-                    .expect("Link path is not an UTF-8 string")
-            );
+    #[cfg(feature = "bundled")]
+    {
+        let mut hwloc2_source_path = PathBuf::from(git_dir);
+        fetch_hwloc(hwloc2_source_path.as_path(), source_version);
+        hwloc2_source_path.push("hwloc");
+        #[cfg(target_os = "windows")]
+        let cmake = hwloc2_source_path.join("contrib").join("windows-cmake").join("CMakeLists.txt").exists();
+        #[cfg(not(target_os = "windows"))]
+        let cmake = false;
+        let mut hwloc2_compiled_path: PathBuf = if cmake {
+            let target = env::var("TARGET").expect("Cargo build scripts always have TARGET");
+            let target_os = get_os_from_triple(target.as_str()).unwrap();
+            hwloc2_source_path.push("contrib");
+            hwloc2_source_path.push("windows-cmake");
+            compile_hwloc2_cmake(hwloc2_source_path.as_path(), target_os)
+        }
+        else {
+            compile_hwloc2_autotools(hwloc2_source_path)
+        };
+
+        //#[cfg(target_os = "windows")]
+        //std::fs::copy(hwloc2_compiled_path.join("lib").join("hwloc.lib").as_path(), hwloc2_compiled_path.join("lib").join("libhwloc.lib").as_path());
+
+        #[cfg(target_os = "windows")]
+        {
+            println!("cargo:rustc-link-lib=static=hwloc");
+            println!("cargo:rustc-link-search={}", hwloc2_compiled_path.join("lib").display());
+        }
+        #[cfg(not(target_os = "windows"))]
+        {
+            env::set_var("PKG_CONFIG_PATH", format!("{}", hwloc2_compiled_path.join("lib").join("pkgconfig").display()));
+            let lib = pkg_config::Config::new()
+                .atleast_version(required_version)
+                .statik(true)
+                .probe("hwloc")
+                .expect("Could not find a suitable version of hwloc");
+        }
+    }
+    #[cfg(not(feature = "bundled"))]
+    {
+        let lib = pkg_config::Config::new()
+            .atleast_version(required_version)
+            .statik(true)
+            .probe("hwloc")
+            .expect("Could not find a suitable version of hwloc");
+        if cfg!(target_family = "unix") {
+            for link_path in lib.link_paths {
+                println!(
+                    "cargo:rustc-link-arg=-Wl,-rpath,{}",
+                    link_path
+                        .to_str()
+                        .expect("Link path is not an UTF-8 string")
+                );
+            }
         }
     }
```

Comment:
This feels redundant. It would be better to outsource the `cfg(not(feature = "bundled"))` version to a `configure_with_pkgconfig()` function and use that fucntion in the `cfg(all(feature = "bundled", not(target_os = "windows")))` case above.